test/integration/core/accounting-oracle-extra-data-full-items.integration.ts

```diff
@@ -0,0 +1,649 @@
+import { ethers } from "hardhat";
+
+import { HardhatEthersSigner } from "@nomicfoundation/hardhat-ethers/signers";
+import { setBalance } from "@nomicfoundation/hardhat-network-helpers";
+
+import {
+  advanceChainTime,
+  ether,
+  EXTRA_DATA_TYPE_EXITED_VALIDATORS,
+  EXTRA_DATA_TYPE_STUCK_VALIDATORS,
+  findEventsWithInterfaces,
+  impersonate,
+  ItemType,
+  LoadedContract,
+  log,
+  prepareExtraData,
+  RewardDistributionState,
+  setAnnualBalanceIncreaseLimit,
+} from "lib";
+import { getProtocolContext, ProtocolContext, withCSM } from "lib/protocol";
+import { reportWithoutExtraData } from "lib/protocol/helpers/accounting";
+import { NOR_MODULE_ID } from "lib/protocol/helpers/nor";
+import { SDVT_MODULE_ID } from "lib/protocol/helpers/sdvt";
+
+import { MAX_BASIS_POINTS, Snapshot } from "test/suite";
+
+import { Burner__factory, NodeOperatorsRegistry } from "../../../typechain-types";
+
+const MAX_NODE_OPERATORS_PER_EXTRA_DATA_ITEM = 24;
+const MAX_ITEMS_PER_EXTRA_DATA_TRANSACTION = 8;
+const CSM_MODULE_ID = BigInt(3);
+
+class ListKeyMapHelper<ValueType> {
+  private map: Map<string, ValueType> = new Map();
+
+  constructor() {
+    this.map = new Map<string, ValueType>();
+  }
+
+  set(keys: unknown[], value: ValueType): void {
+    const compositeKey = this.createKey(keys);
+    this.map.set(compositeKey, value);
+  }
+
+  get(keys: unknown[]): ValueType | undefined {
+    const compositeKey = this.createKey(keys);
+    const result = this.map.get(compositeKey);
+    if (result === undefined) {
+      log.error("HelperMap: get: result is undefined for key " + compositeKey);
+    }
+    return result;
+  }
+
+  private createKey(keys: unknown[]): string {
+    return keys.map((k) => String(k)).join("-");
+  }
+}
+
+describe("Integration: AccountingOracle extra data full items", () => {
+  let ctx: ProtocolContext;
+  let stranger: HardhatEthersSigner;
+
+  let snapshot: string;
+  let originalState: string;
+
+  before(async () => {
+    ctx = await getProtocolContext();
+    snapshot = await Snapshot.take();
+
+    [stranger] = await ethers.getSigners();
+    await setBalance(stranger.address, ether("1000000"));
+
+    if (ctx.isScratch) {
+      const { oracleReportSanityChecker } = ctx.contracts;
+      // Need this to pass the annual balance increase limit check in sanity checker for scratch deploy
+      // with not that much TVL
+      await setAnnualBalanceIncreaseLimit(oracleReportSanityChecker, MAX_BASIS_POINTS);
+
+      // Need this to pass the annual balance increase limit check in sanity checker for scratch deploy
+      // with not that much TVL
+      await advanceChainTime(15n * 24n * 60n * 60n);
+    }
+
+    await prepareModules();
+  });
+
+  beforeEach(async () => (originalState = await Snapshot.take()));
+
+  afterEach(async () => await Snapshot.restore(originalState));
+
+  after(async () => await Snapshot.restore(snapshot));
+
+  async function calcNodeOperatorRewards(
```

Comment:
💭 Move to protocol/helpers? Just to not bloat the file and leave only test inside, to make it more readable?

Reply:
done